doc/en/user/source/production/fips.rst

```diff
@@ -0,0 +1,278 @@
+
+* **BCFKS Keystore Support**: BouncyCastle FIPS KeyStore format for secure key storage
+* **FIPS-Compliant Algorithms**: Cryptographic algorithms that meet FIPS 140-2 requirements
+* **Backward Compatibility**: Support for existing JCEKS keystores with automatic migration
+* **Environment Configuration**: Flexible deployment through environment variables
+
+Enabling FIPS Mode
+------------------
+
+FIPS mode can be enabled through environment variables or system properties:
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~
+
+Set the following environment variables before starting GeoServer:
+
+.. code-block:: bash
+
+   export FIPS_MODE=true
+   export GEOSERVER_KEYSTORE_TYPE=BCFKS
+   export GEOSERVER_KEYSTORE_PROVIDER=BC
+
+System Properties
+~~~~~~~~~~~~~~~~
+
+Alternatively, set Java system properties using the same variable names:
+
+.. code-block:: bash
+
+   java -Dcom.redhat.fips=true \
+        -DGEOSERVER_KEYSTORE_TYPE=BCFKS \
+        -DGEOSERVER_KEYSTORE_PROVIDER=BC \
+        -jar geoserver.war
+
+Docker Container
+~~~~~~~~~~~~~~~
+
+For containerized deployments, use the FIPS-enabled Docker image:
+
+.. code-block:: bash
+
+   docker run -d \
+     -p 8080:8080 \
+     -e FIPS_MODE=true \
+     -e GEOSERVER_KEYSTORE_TYPE=BCFKS \
+     -e GEOSERVER_KEYSTORE_PROVIDER=BC \
+     geoserver-fips:latest
+
+Keystore Configuration
+---------------------
+
+GeoServer supports multiple keystore types for different security requirements:
+
+BCFKS (BouncyCastle FIPS KeyStore)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The BCFKS format provides FIPS 140-2 compliance:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=BCFKS
+   GEOSERVER_KEYSTORE_PROVIDER=BC
+
+JCEKS (Java Cryptography Extension KeyStore)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Traditional Java keystore format for backward compatibility:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=JCEKS
+   GEOSERVER_KEYSTORE_PROVIDER=SunJCE
+
+PKCS12
+~~~~~~
+
+Standard format for certificate and key storage:
+
+.. code-block:: properties
+
+   GEOSERVER_KEYSTORE_TYPE=PKCS12
+   GEOSERVER_KEYSTORE_PROVIDER=BC
+
+Migration from JCEKS to BCFKS
+-----------------------------
+
+GeoServer provides a migration utility to convert existing JCEKS keystores to FIPS-compliant BCFKS format.
+
+Using the Migration Utility
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The migration utility is provided as a standalone script that uses the FIPS utilities JAR:
+
+.. code-block:: bash
+
+   ./migrate-keystore.sh -s source.jks -sp oldpass -t target.bcfks -tp newpass
+
+The script automatically handles:
+* Dependency checking (BouncyCastle JARs)
+* File validation and permissions
+* Auto-detection of keystore types and providers
+* Error handling and colored output
+
+Parameters:
+
+* ``-s, --source``: Source keystore file path
+* ``-sp, --source-password``: Source keystore password
+* ``-t, --target``: Target keystore file path
+* ``-tp, --target-password``: Target keystore password
```

Comment:
will the target keystore be created if it doesn't exist? Should make a note of that here.